opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1963,6 +1965,55 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
         return m_active_to_global;
     }
 
+    void EclipseGrid::init_lgr_cells(const LgrCollection& lgr_input) {
+        create_lgr_cells_tree(lgr_input);
+    }
+
+    void EclipseGrid::create_lgr_cells_tree(const LgrCollection& lgr_input) {   
+         auto IJK_global = [](const auto& lgr_cell, const auto& i_list, const auto& j_list, const auto& k_list){
+            if ((i_list.size() != j_list.size()) || (j_list.size() != k_list.size()) || (k_list.size() != i_list.size()) ){
+                 throw std::invalid_argument("Sizes are not compatible.");
+            }
+            std::vector<std::size_t> global_ind(i_list.size());
```

Comment:
Maybe it's the same when you do std::vector<std::size_t> global_ind(i_list.size(), 1) and avoid the for-loop?